lib/archethic/mining/validation_context.ex

```diff
@@ -398,45 +393,107 @@ defmodule Archethic.Mining.ValidationContext do
          node_public_key
        )
        when is_binary(node_public_key) do
-    Enum.any?(stamps, &(&1.node_public_key == node_public_key))
+    Enum.any?(stamps, fn
+      {^node_public_key, _} -> true
+      _ -> false
+    end)
   end
 
   @doc """
   Determines if a node is a cross validation node
 
   ## Examples
 
-    iex> %ValidationContext{
-    ...>   coordinator_node: %Node{last_public_key: "key1"},
-    ...>   cross_validation_nodes: [
-    ...>     %Node{last_public_key: "key2"},
-    ...>     %Node{last_public_key: "key3"},
-    ...>     %Node{last_public_key: "key4"}
-    ...>   ]
-    ...> }
-    ...> |> ValidationContext.cross_validation_node?("key3")
-    true
-
-    iex> %ValidationContext{
+    iex> context = %ValidationContext{
     ...>   coordinator_node: %Node{last_public_key: "key1"},
     ...>   cross_validation_nodes: [
     ...>     %Node{last_public_key: "key2"},
     ...>     %Node{last_public_key: "key3"},
     ...>     %Node{last_public_key: "key4"}
+    ...>   ],
+    ...>   chain_storage_nodes: [
+    ...>     %Node{last_public_key: "key5"},
+    ...>     %Node{last_public_key: "key6"},
+    ...>     %Node{last_public_key: "key7"}
     ...>   ]
     ...> }
-    ...> |> ValidationContext.cross_validation_node?("key1")
+    ...> 
+    ...> ValidationContext.cross_validation_node?(context, "key3")
+    ...> true
+    ...> ValidationContext.cross_validation_node?(context, "key7")
+    ...> true
+    ...> ValidationContext.cross_validation_node?(context, "key1")
     false
+
   """
   @spec cross_validation_node?(t(), Crypto.key()) :: boolean()
   def cross_validation_node?(
```

Comment:
Suggestion: rename the function somehow

Reply:
In fact the function name is valid, we receive a cross validation stamp and assert it comes from a cross validation node so to me the name is valid